### PR TITLE
Core: Parse floats in epoch millis parser

### PR DESCRIPTION
**Pull request by spinscale**
In order to stay BWC compatible with joda time, the epoch millis date
formatter needs to properly parse dates with a dot like `123.45`. This
adds this functionality for the epoch millis parser in the same way as
for the epoch seconds parser.

### CODE REVIEW

**Review by cbuescher — CHANGES_REQUESTED** [review]
@spinscale I left a short question, otherwise I think this looks good.

File: server/src/main/java/org/elasticsearch/common/time/EpochMillisDateFormatter.java [hunk 1]
```diff
@@ -39,20 +41,42 @@
...
+                if (inputs[1].length() == 0) {
+                    // this is BWC compatible to joda time, nothing after the dot is allowed
+                    return Instant.ofEpochMilli(milliSeconds).atZone(ZoneOffset.UTC);
+                }
+                // scientific notation it is!
+                if (inputs[1].contains("e")) {
+                    return Instant.ofEpochMilli(Double.valueOf(input).longValue()).atZone(ZoneOffset.UTC);
+                }
+
+                if (inputs[1].length() > 9) {
```
[the remark this answers is not in the document]
Comment (spinscale) [reply server/src/main/java/org/elasticsearch/common/time/EpochMillisDateFormatter.java]: totally. fixed and tested...

I also fixed the negative handling of the epoch seconds parsing which was broken and added a test

**Review by cbuescher — APPROVED** [review]
@spinscale  Thanks, LGTM

**Comment by spinscale** [comment]
@elasticmachine retest this please

**Pull request merged**